### ISSUE TITLE
Bug: [QueryBuilder] Like function of Database library fails when used with Concat

### ISSUE THREAD

**Issue by mmfarhan** [codeigniter4/CodeIgniter4#3970]
Hi,

it looks like there is some issue with the like function in db library:
`("concat(users.name, ' ', IF(users.surname IS NULL or users.surname = '', '', users.surname)) full_name")` works in the select statement but fails in the like function.
It produces this:
`OR  concat(users.name, ' ', IF(users.surname IS NULL or `users`.`surname` = '', '', users.surname))  LIKE :concat(users.name, ' ', IF(users.surname IS NULL or `users`.`surname` = '', '', users.surname)): ESCAPE '!'`

This simple code will result into error:
```php
function sqlTest(){
		$search['value'] = 'perfect';
		$usersModel = new \App\Models\Users_model();
		$usersModel
				->select("concat(users.name, ' ', IF(users.surname IS NULL or users.surname = '', '', users.surname)) full_name")
				->groupStart()
				->orLike("users.email", $search['value'], 'both')
				->orLike("users.phone", $search['value'], 'both')
				->orLike("concat(users.name, ' ', IF(users.surname IS NULL or users.surname = '', '', users.surname))", $search['value'], 'both')
				->groupEnd()
				->findAll();
		
		echo $usersModel->getLastQuery();
	}
```

**Comment by najdanovicivan** [comment]
Try using havingLike() for concatenate value

**Comment by kenjis** [comment]
https://codeigniter4.github.io/CodeIgniter4/database/query_builder.html#query-builder-like-rawsql
Fixed by #5817
